### DISCUSSION TITLE
Mandatory --override...?

### DISCUSSION THREAD

**Discussion by cgerke** [microsoft/winget-pkgs#97552] [Ideas] [answered]
How would everyone feel about adding functionality that specified an override is MANDATORY  for apps that require serials if attempting silent installs.

This would be for scenarios where vendor software requires a serial number when attempting a silent install. For example SigmaPlot by SystatSoftwareInc

winget install --manifest C:\Users\gerkec\winget-pkgs\manifests\s\SystatSoftwareInc\SigmaPlot\14.5\ --silent --override "/S SERIALNUMBER=XXXXXXXXX INSTALLPATH=c:\ProgramData\SigmaPlot\SPW14 SILENT=TRUE ALLUSERS=TRUE USERNAME=ADMIN"

**Comment by denelon** [comment, marked as answer]
Do you have any other examples of packages with this behavior? We added a post-install notes field. My big concern here is that we wouldn't necessarily have a serial number to test with in our validation environment.

**Reply by cgerke** [reply]
Good point. In all my overrides that I’ve used that require this behaviour it’s mostly serial numbers or license server addresses. Since it’s very much enterprise centric it’s probably best handle by post install scripts or by wrapping winget.
